DataFormats/PatCandidates/interface/Jet.h

```diff
@@ -498,21 +500,33 @@ namespace pat {
 
 
       /// String access to subjet list
-      pat::JetPtrCollection const & subjets( std::string label ) const ;
+      pat::JetPtrCollection const & subjets( std::string const & label ) const ;
 
       /// Add new set of subjets
-      void addSubjets( pat::JetPtrCollection const & pieces, std::string label = ""  );
+      void addSubjets( pat::JetPtrCollection const & pieces, std::string const & label = ""  );
 
       /// Check to see if the subjet collection exists
-      bool hasSubjets( std::string label ) const { return find( subjetLabels_.begin(), subjetLabels_.end(), label) != subjetLabels_.end(); }
+      bool hasSubjets( std::string const & label ) const { return find( subjetLabels_.begin(), subjetLabels_.end(), label) != subjetLabels_.end(); }
       
       /// Number of subjet collections
       unsigned int nSubjetCollections(  ) const { return  subjetCollections_.size(); }
       
       /// Subjet collection names
       std::vector<std::string> const & subjetCollectionNames() const { return subjetLabels_; }
 
-
+      /// Access to mass of subjets
+      double groomedMass(unsigned int index = 0) const{
+	return nSubjetCollections() > 0 && subjets(0).size() > 0 ?
+	  std::accumulate( subjets(0).begin(), subjets(0).end(),
```

Comment:
I didn't notice it in the previous iterations, but I suppose the `(0)` is meant to be `(index)`, right?

scram b code-checks may complain about using `x.size() > 0` instead of `!x.empty()` (here and in the next line), maybe you can change that as well?